### ISSUE TITLE
Not allow to call as function

### ISSUE THREAD

**Issue by Paxa** [vitaly-t/connection-string#24]
Would be nice to have error when we call it as a function
```js
var ConnectionString = require('connection-string').ConnectionString;
ConnectionString('postgres://localhost:5432/my_db');
=> 
TypeError: Cannot set property 'protocol' of undefined
    at ConnectionString (/Users/me/app/node_modules/connection-string/dist/main.js:22)
```

**Comment by vitaly-t** [comment]
In JavaScript, when you call something starting with the capital, that means it is a class name, which requires that `new` is used.

And pretty much no library will attempt provide extra handling when you are misusing a class. So why would you expect that library to do so?

**Comment by Paxa** [comment]
I know we should read docs better but sometimes forget `new` and got weird error

If generated code used `class ConnectionString { .... }` then it would throw an error
```
Uncaught TypeError: Class constructor ConnectionString cannot be invoked without 'new'
```
But generated code uses `function ConnectionString` inside unreadable mess.

To make it work as normal JS class we can add here https://github.com/vitaly-t/connection-string/blob/4d2166eac24d5ea7b1ea27440293ddc24cf9c5ac/src/main.ts#L53
```js
if (!(this instanceof ConnectionString)) {
  throw new TypeError("Class constructor ConnectionString cannot be invoked without 'new'")
}
```

**Comment by vitaly-t** [comment]
@Paxa 

Added implementation, plus a test, but for some strange reasons, the test, even though it definitely runs, does not get included into the coverage. I cannot understand why that is. Any idea?

**Comment by vitaly-t** [comment]
Implemented in [v3.0.5](https://github.com/vitaly-t/connection-string/releases/tag/3.0.5).

**Comment by Paxa** [comment]
Blagodaru
